File: ROS_service/src/ros_service/src/add_two_ints_client.py
#!/usr/bin/env python

import rospy
from addTwoInts.srv import AddTwoInts, AddTwoIntsRequest

def add_two_ints_client(x, y):
    rospy.wait_for_service('add_two_ints')
    try:
        add_two_ints = rospy.ServiceProxy('add_two_ints', AddTwoInts)
        response = add_two_ints(x, y)
        return response.sum
    except rospy.ServiceException as e:
        print("Service call failed: {}".format(e))

if __name__ == "__main__":
    x = 5
    y = 3
    print("Requesting {} + {}...".format(x, y))
    result = add_two_ints_client(x, y)
    print("Result: {} + {} = {}".format(x, y, result))

File: ROS_service/src/ros_service/src/add_two_ints_client.cc
#include "ros/ros.h"
#include "addTwoInts/AddTwoInts.h"

int main(int argc, char **argv)
{
    ros::init(argc, argv, "add_two_ints_client");
    if (argc != 3)
    {
        ROS_INFO("Usage: add_two_ints_client X Y");
        return 1;
    }

    ros::NodeHandle n;
    ros::ServiceClient client = n.serviceClient<addTwoInts::AddTwoInts>("add_two_ints");
    addTwoInts::AddTwoInts srv;
    srv.request.a = atoll(argv[1]);
    srv.request.b = atoll(argv[2]);
    if (client.call(srv))
    {
        ROS_INFO("Sum: %ld", (long int)srv.response.sum);
    }
    else
    {
        ROS_ERROR("Failed to call service add_two_ints");
        return 1;
    }

    return 0;
}
